### PR TITLE
add vertical pod autoscaler

### PR DESCRIPTION
**Pull request by QuentinBisson**
Towards: https://github.com/giantswarm/roadmap/issues/358

This PR adds a vertical pod autoscaler to alloy and sets some sane defaults resource usage based on the alloy-rules usage:

image

### Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Comment by QuentinBisson** [comment]
@giantswarm/team-atlas ready for review

File: helm/alloy/templates/vertical-pod-autoscaler.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+### Upstream work being tracked at https://github.com/grafana/alloy/pull/1305
+{{ if .Capabilities.APIVersions.Has "autoscaling.k8s.io/v1" -}}
```
Comment (QuantumEnigmaa) [inline helm/alloy/templates/vertical-pod-autoscaler.yaml:2]: I think we don't even need this as Alloy will only be deployed on clusters using recent releases (as of now) which all have the VPA CRDs embedded.

Comment (QuentinBisson) [reply helm/alloy/templates/vertical-pod-autoscaler.yaml:2]: True but this will be upstream anyway :D

Comment (QuantumEnigmaa) [reply helm/alloy/templates/vertical-pod-autoscaler.yaml:2]: Oh ok nevermind then :)

**Review by QuantumEnigmaa — APPROVED** [review]
LGTM if tested

**Pull request merged**